src/verilog/verilog_typecheck.cpp

```diff
@@ -1558,11 +1621,23 @@ void verilog_typecheckt::convert_statements()
   verilog_module_expr.module_items().reserve(module_items.get_sub().size());
 
   // do the generate stuff, copying the module items
-
   for(auto &item : module_items.get_sub())
     elaborate_generate_item(
       static_cast<const exprt &>(item), verilog_module_expr.module_items());
 
+  // Do defparam, also known as 'parameter override'.
+  // These must all be done before any module instantiation,
+  // which use the parameters.
+  for(auto &item : verilog_module_expr.module_items())
+  {
+    if(item.id() == ID_parameter_override)
+      convert_parameter_override(to_verilog_parameter_override(item));
+    else if(item.id() == ID_set_genvars)
+      for(auto &sub_item : item.operands())
+        if(sub_item.id() == ID_parameter_override)
+          convert_parameter_override(to_verilog_parameter_override(sub_item));
```

Comment:
Would appreciate braces around the multi-line bodies here.

Reply:
done